### PR TITLE
Change Ft4222 to work with multiple I2C devices (I2cBus prototype)

### PR DESCRIPTION
**Pull request by krwq**
Testing idea of I2C bus from: https://github.com/dotnet/iot/issues/1335

also fixing Ft4222 to work with multiple devices. What I just learned is that SenseHat is using same I2cDevice address for both Joystick and LED matrix but it uses 2 devices.

### CODE REVIEW

File: src/devices/Ft4222/samples/Program.cs [hunk 1]
```diff
@@ -31,13 +31,21 @@
     Console.WriteLine($"Device type: {device.Type}");
 }
 
+if (devices.Count == 0)
+{
+    Console.WriteLine("No devices to connected to run tests.");
```
Comment (krwq) [inline src/devices/Ft4222/samples/Program.cs:36]:
```suggestion
    Console.WriteLine("No devices connected to run tests.");
```

File: src/devices/Ft4222/samples/Program.cs [hunk 2]
```diff
@@ -55,11 +63,11 @@
...
 
-void TestI2c()
+void TestI2c(DeviceInformation device)
 {
-    using Ft4222I2c ftI2c = new(new I2cConnectionSettings(0, Bno055Sensor.DefaultI2cAddress));
-
-    Bno055Sensor bno055Sensor = new(ftI2c);
+    using Ft4222I2cBus ftI2c = new(device);
+    using I2cDevice i2cDevice = ftI2c.CreateDevice(Bno055Sensor.DefaultI2cAddress);
+    Bno055Sensor bno055Sensor = new(i2cDevice);
```
Comment (Ellerbach) [inline src/devices/Ft4222/samples/Program.cs:70]: You should add how to create multiple devices maybe. Just to show it's possible

File: src/devices/Ft4222/Ft4222I2cBus.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.Ft4222
+{
+    /// <summary>
+    /// FT4222 I2C Device
+    /// </summary>
+    public class Ft4222I2cBus : IDisposable
```
Comment (Ellerbach) [inline src/devices/Ft4222/Ft4222I2cBus.cs:16]: So related to the issue about I2cBus, should you directly then propose an I2cBus implementation and use FT4222 as an example of implementation?
On what I can see here, you have a good list of functions :-)

**Review by Ellerbach — APPROVED** [review]
Seems great! And a good base to setup I2cBus then!

**Comment by pgrawehr** [comment]
@krwq: The build is broken.

File: src/devices/Ft4222/Ft4222I2cBus.cs (hunk 3 above)
Comment (krwq) [reply src/devices/Ft4222/Ft4222I2cBus.cs:16]: do not want to do this as part of this PR

**Pull request merged**